beta/nncf/tensorflow/pruning/schedulers.py

```diff
@@ -0,0 +1,151 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import numpy as np
+
+from nncf.api.compression import CompressionScheduler
+from nncf.common.utils.registry import Registry
+
+PRUNING_SCHEDULERS = Registry("pruning_schedulers")
+
+
+class PruningScheduler(CompressionScheduler):
+    def __init__(self, pruning_algo, params: dict = None):
+        super().__init__()
+        if params is None:
+            self._params = dict()
+        else:
+            self._params = params
+
+        self.algo = pruning_algo
+        self.initial_pruning_rate = self.algo.pruning_init
+        self.pruning_target = self._params.get('pruning_target', 0.5)
+        self.pruning_steps = self._params.get('pruning_steps', 100)
+        self.pruning_freeze_epoch = self.pruning_steps
+
+    def _calc_density_level(self):
+        return 1 - self.current_pruning_level
+
+    def _maybe_freeze(self):
+        if self.current_epoch + 1 >= self.pruning_freeze_epoch:
+            self.algo.freeze()
+
+    @property
+    def current_pruning_level(self):
+        raise NotImplementedError
+
+    @property
+    def target_pruning_level(self):
+        return self.pruning_target
+
+    def load_state(self, state):
+        super().load_state(state)
+        self._maybe_freeze()
+
+
+@PRUNING_SCHEDULERS.register("polynomial")
+class PolynomialSparseScheduler(PruningScheduler):
```

Comment:
Can at this stage the common schedulers be used already?

Reply:
@andrey-churkin please comment on this.

Reply:
I suppose @andrey-churkin will align the pruning schedulers usage in one of his upcoming PRs, then.